app/Http/Controllers/Backend/Social/MastodonController.php

```diff
@@ -123,21 +122,99 @@ private static function updateToken(User $user, SocialiteUser $socialiteUser, Ma
                                      ]);
     }
 
-    public static function postStatus(Status $status): void {
-        if ($status?->user?->socialProfile?->mastodon_server === null || config('trwl.post_social') !== true) {
+    public static function postStatus(Status $status, bool $shouldPostAsChain = false): void {
+        if (config('trwl.post_social') !== true) {
+            Log::error("Was dispatched to post on Mastodon, but POST_SOCIAL env variable is not set.");
             return;
         }
 
+        if ($status?->user?->socialProfile?->mastodon_server === null) {
+            return;
+        }
+
+        $traverseChain = null;
+        if ($shouldPostAsChain) {
+            $chainHead = self::getLastSavedPostIdFromUserStatuses($status->user);
+
+            if ($chainHead != null) {
+                $traverseChain = self::getEndOfChain($status->user, $chainHead->mastodon_post_id);
+            }
+        }
+
         try {
             $statusText     = $status->socialText . ' ' . url("/status/{$status->id}");
             $mastodonDomain = MastodonServer::find($status->user->socialProfile->mastodon_server)->domain;
             Mastodon::domain($mastodonDomain)->token($status->user->socialProfile->mastodon_token);
-            Mastodon::createStatus($statusText, ['visibility' => 'unlisted']);
+
+            $post_response = Mastodon::createStatus($statusText, [
+                'visibility'     => 'unlisted',
+                'in_reply_to_id' => $traverseChain
+            ]);
+
+            $status->update(['mastodon_post_id' => $post_response['id']]);
             Log::info("Posted on Mastodon (domain=" . $mastodonDomain . "): " . $statusText);
-        } catch (RequestException $e) {
-            $status->user->notify(new MastodonNotSent($e->getResponse()?->getStatusCode(), $status));
+        } catch (GuzzleException $e) {
+            $status->user->notify(new MastodonNotSent($e->getCode(), $status));
         } catch (Exception $e) {
             Log::error($e);
         }
     }
+
+    /**
+     * @param User   $user             Trwl user with social profile
+     * @param string $mastodon_post_id Id of a known post of the chain, taken from the statuses table.
+     *
+     * @return string|null The id of the last post that can be replied to. Returns null if there is none.
+     * @see https://docs.joinmastodon.org/entities/Context/ Documentation of the Mastodon Context API.
+     * @see https://docs.joinmastodon.org/methods/statuses/ Documentation of the Mastodon Status API.
+     */
+    public static function getEndOfChain(User $user, string $mastodon_post_id): ?string {
```

Comment:
is `string` really correct?

Reply:
Yes. Mastodon sends the ids as string afaik, and we save in the database as such too.